server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -332,6 +332,11 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
 
             final Nested nested;
             if (nestedSort != null) {
+                if ((context.indexVersionCreated().before(Version.V_6_5_0) || nestedSort.getNestedSort() != null) &&
+                    nestedSort.getMaxChildren() != Integer.MAX_VALUE) {
+                    throw new QueryShardException(context,
```

Comment:
the error message should be different if `nestedSort.getNestedSort() != null` since it's not an issue with the version of the index